specs/destination.go

```diff
@@ -43,10 +40,23 @@ func (d *Destination) UnmarshalSpec(out interface{}) error {
 	if err != nil {
 		return err
 	}
-	dec := json.NewDecoder(nil)
+	dec := json.NewDecoder(bytes.NewReader(b))
 	dec.UseNumber()
 	dec.DisallowUnknownFields()
-	return json.Unmarshal(b, out)
+	return dec.Decode(out)
+}
+
+func (d *Destination) Validate() error {
+	if d.Name == "" {
+		return fmt.Errorf("name is required")
+	}
+	if d.Version == "" {
+		return fmt.Errorf("version is required")
+	}
+	if !strings.HasPrefix(d.Version, "v") {
```

Comment:
Nit, we should do full SemVer validation as the following string `vv1.0.0` passes this validation